lib/rational/src/Rational.cpp

```diff
@@ -9,22 +9,28 @@ Rational::Rational(int numerator, int denominator)
 	, denominator(denominator)
 {}
 
-Rational Rational::operator+(const Rational& other) {
+Rational& Rational::operator+=(const Rational& other) {
 	Rational a = (*this).reduce();
 	Rational b = other.reduce();
-	
+
 	int multiple = lcm(a.denominator, b.denominator);
 
 	a.numerator *= multiple / a.denominator;
 	b.numerator *= multiple / b.denominator;
 
-	return Rational(
+	Rational rez(
 		a.numerator + b.numerator,
 		multiple
-	).reduce();
+	);
+
+	rez.reduce();
+	this->numerator = rez.numerator;
+	this->denominator = rez.denominator;
+
+	return rez;
```

Comment:
It's mistaken because a reference to a local object is returned, the object would be destroyed immediately after return. Here you should return *this.
Need to check others methods.